src/shared/api/schemas.ts

```diff
@@ -0,0 +1,308 @@
+    z.object({ kind: z.literal('Store'), metadata: metadataSchema }),
+  ]),
+});
+
+export type Asset = z.infer<typeof assetSchema>;
+
+export const assetDefinitionSchema = z.object({
+  id: z
+    .string()
+    .superRefine((val, ctx) => {
+      if (val.split('#').length !== 2) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.invalid_literal,
+          message: 'Invalid assetDefinitionId format',
+          expected: 'asset#asset_domain format',
+          received: val,
+        });
+      }
+    })
+    .transform(transformToAssetDefinitionId),
+  type: z.enum(['Numeric', 'Store']),
+  logo: z.string().nullable(),
+  assets: z.number(),
+  metadata: metadataSchema,
+  mintable: z.enum(['Infinitely', 'Once', 'Not']),
+  owned_by: z
+    .string()
+    .superRefine((val, ctx) => {
+      if (val.split('@').length !== 2) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.invalid_literal,
+          message: 'Invalid accountId format',
+          expected: 'signatory@account_domain format',
+          received: val,
+        });
+      }
+    })
+    .transform(transformToAccountId),
+});
+
+export type AssetDefinition = z.infer<typeof assetDefinitionSchema>;
+
+export const domainSchema = z.object({
+  id: z.string().transform((ctx) => new DomainId(ctx)),
+  logo: z.string().nullable(),
+  metadata: metadataSchema,
+  owned_by: z.string(),
+  accounts: z.number(),
+  assets: z.number(),
+});
+
+export type Domain = z.infer<typeof domainSchema>;
+
+export interface TransactionSearchParams extends PaginationParams {
+  authority?: string
+  block_hash?: string
+}
+
+export const transactionSchema = z.object({
+  authority: z.string(),
+  hash: z.string(),
+  block_hash: z.string(),
+  created_at: z.string().transform((x) => new Date(x)),
+  instructions: z.enum(['Instructions', 'Wasm']),
+  status: z.enum(['Committed', 'Rejected']),
+});
+
+export const detailedTransactionSchema = z.object({
+  authority: z.string(),
+  hash: z.string(),
+  block_hash: z.string(),
+  created_at: z.string().transform((x) => new Date(x)),
+  instructions: z.enum(['Instructions', 'Wasm']),
+  status: z.enum(['Committed', 'Rejected']),
+  rejection_reason: z.record(z.string(), z.any()).nullable(),
+  metadata: metadataSchema,
+  nonce: z.number().nullable(),
+  signature: z.string(),
+  time_to_live: z
+    .object({
+      ms: z.number().min(0),
+    })
+    .nullable(),
+});
+
+export type Transaction = z.infer<typeof transactionSchema>;
+export type DetailedTransaction = z.infer<typeof detailedTransactionSchema>;
+
+export const blockSchema = z.object({
+  hash: z.string(),
+  height: z.number(),
+  created_at: z.string().transform((x) => new Date(x)),
+  prev_block_hash: z.string().nullable(),
+  transactions_hash: z.string(),
+  transactions_rejected: z.number(),
+  transactions_total: z.number(),
+  consensus_estimation: z.object({
+    ms: z.number().min(0),
+  }),
```

Comment:
I would suggest making a separate `Duration` schema, as it might change in future and is used in several places.

```suggestion
  consensus_estimation: Duration,
```

Same for `Date`.